crates/forge-runner/src/lib.rs

```diff
@@ -292,19 +292,75 @@ fn choose_test_strategy_and_run(
     runner: Arc<SierraCasmRunner>,
     runner_config: Arc<RunnerConfig>,
     runner_params: Arc<RunnerParams>,
-    send: &Sender<()>,
-) -> JoinHandle<Result<TestCaseSummary>> {
+    send: Sender<()>,
+) -> JoinHandle<Result<AnyTestCaseSummary>> {
     if args.is_empty() {
-        run_test(case, runner, runner_config, runner_params, send.clone())
+        tokio::task::spawn(async move {
+            let res = run_test(case, runner, runner_config, runner_params, send).await??;
+            Ok(AnyTestCaseSummary::Single(res))
+        })
     } else {
-        run_with_fuzzing(
-            args,
-            case,
-            runner,
-            runner_config,
-            runner_params,
-            send.clone(),
-        )
+        tokio::task::spawn(async move {
+            let res =
+                run_with_fuzzing(args, case, runner, runner_config, runner_params, send).await??;
+            Ok(AnyTestCaseSummary::Fuzzing(res))
+        })
+    }
+}
+
+fn fuzzing_run_summary(results: Vec<TestCaseSummary<Single>>) -> TestCaseSummary<Fuzzing> {
```

Comment:
Can this be done in a more idiomatic manner? like `TestCaseSummary.from(vec_of_singles)` ?

Reply:
Yes! Good call!

Reply:
Done